### PR TITLE
Support multiple channels

### PR DESCRIPTION
**Pull request by invis-z**
This is an amateur's attempt to support multiple channels. I implemented it by taking an average of FFT results across all channels since blindly adding raw samples might cause phase issues.

Tested on Windows after built in WSL, seems to be working.

I don't know if this should count as bugfix since it is technically a new feature. But this does make audio files with only the right channel work with this program, although they are and should be very rare.

Thanks!

### CODE REVIEW

**Comment by rexim** [comment]
How did you come up with this approach? Is this how it is usually done in frequency visualizers?

**Comment by invis-z** [comment]
To be honest, I don't actually know about anything regarding what is the right way to deal with this problem. 

The other day I watched a video mentioning playing music in mono resulting in degraded audio quality caused by the phase difference between the channels cancelling part of the sound and amplifying other part of it. Although this usually only happens in very old audio mixings, that video makes me think that just blindly averaging all the channel together can be a bad idea, so I tried to go to the next logical option, which is deal with it after FFT. 

To me, this approach makes sense logically. It seems to be closer to what the listening experience would be in real life with a multi-speaker setup, and all the sound "adds" together. But it also might just be my imagination. Also, this is the only approach that falls within my programming ability. So I tried it with a song in both channels and stereo and visually it seems to be ok.

TL; DR, I pulled it out from thin air and have no idea if it is any good. So please just take this as an idea, perhaps a bad one.

**Comment by rexim** [comment]
I see, I see. I also don't know how it should be done properly. I need to do a bit more research on that.

**Comment by rexim** [comment]
I'm still not sure how to go about it. I created a TODO to not forget about it c3f16b269efa700844e06b044064c5c5ef902402. I'll keep this at the back of my head for the time being. Thank you for your PR!
